src/ai/backend/manager/container_registry/aws_ecr.py

```diff
@@ -43,10 +48,12 @@ async def fetch_repositories(
                 for repo in response["repositories"]:
                     match type_:
                         case "ecr":
-                            yield repo["repositoryName"]
+                            if repo["repositoryName"].startswith(self.registry_info.project):
+                                yield repo["repositoryName"]
                         case "ecr-public":
                             registry_alias = (repo["repositoryUri"].split("/"))[1]
-                            yield f"{registry_alias}/{repo["repositoryName"]}"
+                            if self.registry_info.project == registry_alias:
+                                yield f"{registry_alias}/{repo["repositoryName"]}"
```

Comment:
What happens if the conditional `startswith` doesn't apply here?
`if repo["repositoryName"].startswith(self.registry_info.project)` and `if self.registry_info.project == registry_alias`

Reply:
It is not problematic images not belonging to the project are not rescanned.

The previous behavior is rescanning all images but committing only related images.
Now, it only rescans images related to the project.